src/components/ha-selector/ha-selector-location.ts

```diff
@@ -51,8 +51,14 @@ export class HaLocationSelector extends LitElement {
       return [
         {
           id: "location",
-          latitude: value?.latitude || this.hass.config.latitude,
-          longitude: value?.longitude || this.hass.config.longitude,
+          latitude:
+            !value || isNaN(value.latitude)
+              ? this.hass.config.latitude
+              : value.latitude,
+          longitude:
+            !value || isNaN(value.longitude)
+              ? this.hass.config.longitude
+              : value.longitude,
           radius: selector.location?.radius ? value?.radius || 1000 : undefined,
```

Comment:
I see you didn't touch the radius. There is an [issue outstanding](https://github.com/home-assistant/core/issues/101301) where if the radius is required (not optional) by a config_flow and the radius is not changed from the default radius during that config_flow, then an error `User input malformed: expected float for dictionary value @ data['location']['radius']` will be generated. Do you think that you might be able to fix this too? As far as I can tell geo_json_events is the only integration that is using the mandatory radius which is why it maybe wasn't picked up till now.

Reply:
I would be open to looking at additional fixes for location selector, but I think it would be good if there was a frontend ticket for the issue (so I won't forget). The one in core is probably not in the right place.